### PR TITLE
remarkable-toolchain: use which instead of patching with sed

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/misc/remarkable/remarkable-toolchain/default.nix [hunk 1]
```diff
@@ -1,36 +1,34 @@
...
-    python3
+    python
     file
+    which
   ];
 
-  unpackCmd = "mkdir src; install $curSrc src/install-toolchain.sh";
-
-  dontBuild = true;
+  phases = [ "patchPhase" "installPhase" ];
```
Comment (NickHu) [inline pkgs/development/tools/misc/remarkable/remarkable-toolchain/default.nix:20]: I think overriding `phases` is considered a bad idea

**Review by SuperSandro2000 — COMMENTED** [review]
I don't think the commit title matches that well but I don't any better right now.

**Comment by SuperSandro2000** [comment]
Fails to build with 

```
unpacking sources
unpacking source archive /nix/store/rwnpam335zr38wdx5kv00w54d89l7fvh-oecore-x86_64-cortexa9hf-neon-toolchain-zero-gravitas-1.8-23.9.2019.sh
do not know how to unpack source archive /nix/store/rwnpam335zr38wdx5kv00w54d89l7fvh-oecore-x86_64-cortexa9hf-neon-toolchain-zero-gravitas-1.8-23.9.2019.sh
```

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 102136` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>remarkable-toolchain</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102136` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>remarkable-toolchain</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 102136` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)

**Comment by FRidh** [comment]
What's the motivation for switching to Python 2?

**Pull request merged**